scripts/pre_push_hook.py

```diff
@@ -55,6 +55,30 @@
 LINTER_FILE_FLAG = '--files'
 PYTHON_CMD = 'python'
 FRONTEND_TEST_SCRIPT = 'run_frontend_tests.sh'
+GIT_IS_DIRTY_CMD = 'git status --porcelain --untracked-files=no'
+
+
+class ChangedBranch(object):
+    def __init__(self, new_branch):
+        get_branch_cmd = 'git symbolic-ref -q --short HEAD'.split()
+        self.old_branch = subprocess.check_output(get_branch_cmd).strip()
+        self.new_branch = new_branch
+        self.is_same_branch = self.old_branch == self.new_branch
+
+    def __enter__(self):
+        if not self.is_same_branch:
+            try:
+                subprocess.check_output(['git', 'checkout', self.new_branch])
+            except subprocess.CalledProcessError:
+                print ('\nCould not change to %s. This is most probably because'
+                       ' you are in a dirty state. Change to the branch that is'
```

Comment:
Hm, it might be worth flagging that in the error message (if you see this message regularly, something has gone wrong, please contact the developers)?